crates/compiler/test_gen/src/helpers/wasm_test_platform.c

```diff
@@ -64,6 +66,7 @@ void *roc_alloc(size_t size, unsigned int alignment)
         ASSERT(num_alloc <= rc_pointers_capacity, "Too many allocations %zd > %zd", num_alloc, rc_pointers_capacity);
 
         size_t *rc_ptr = alloc_ptr_to_rc_ptr(allocated, alignment);
+        *rc_ptr = RC_LIST_INDICATOR;
```

Comment:
There is something non-obvious happening here so I think it's definitely worth adding a code comment. (When you get to it. I know it's still WIP at the moment.)
Are we going to require all roc_alloc implementations to do this?
Also, I notice it is happening unconditionally for every allocation, not just lists. I assume it will get overridden for other types? But why isn't there some Roc code somewhere to initialise the refcount for the list?

Reply:
Yeah, this is totally a hack that I should document better. It is so that we can figure out which allocations are lists and get their refcount. No other platforms will need this. It is strictly need for the introspection without knowing the underlying data type. I'll add some comments.

Reply:
Ok, removed that hack and just make it explicit in the tests:
```rust
    assert_refcounts!(
        indoc!(
            r#"
                s = Str.concat "A long enough string " "to be heap-allocated"
                list = [s, s, s]
                [list, list]
            "#
        ),
        RocList<RocList<RocStr>>,
        &[
            (StandardRC, Live(3)), // s
            (AfterSize, Live(2)),  // list
            (AfterSize, Live(1))   // result
        ]
    );
```